apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/Mintable.tsx

```diff
@@ -0,0 +1,477 @@
+            </AccordionItem>
+
+            <AccordionItem
+              value="primary-sale-recipient"
+              className="border-none "
+            >
+              <AccordionTrigger className="border-border border-t px-1">
+                Primary Sale Recipient
+              </AccordionTrigger>
+              <AccordionContent className="px-1">
+                <PrimarySalesSection
+                  isOwnerAccount={props.isOwnerAccount}
+                  primarySaleRecipient={props.primarySaleRecipient}
+                  update={props.updatePrimaryRecipient}
+                />
+              </AccordionContent>
+            </AccordionItem>
+          </Accordion>
+        </div>
+      )}
+    </ModuleCardUI>
+  );
+}
+
+const primarySaleRecipientFormSchema = z.object({
+  primarySaleRecipient: addressSchema,
+});
+
+function PrimarySalesSection(props: {
+  primarySaleRecipient: string | undefined;
+  update: (values: UpdateFormValues) => Promise<void>;
+  isOwnerAccount: boolean;
+}) {
+  const form = useForm<UpdateFormValues>({
+    resolver: zodResolver(primarySaleRecipientFormSchema),
+    values: {
+      primarySaleRecipient: props.primarySaleRecipient ?? "",
+    },
+  });
+
+  const updateMutation = useMutation({
+    mutationFn: props.update,
+  });
+
+  const onSubmit = async () => {
+    const promise = updateMutation.mutateAsync(form.getValues());
+    toast.promise(promise, {
+      success: "Successfully updated primary sale recipient",
+      error: (error) => `Failed to update primary sale recipient: ${error}`,
+    });
+  };
+
+  return (
+    <Form {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)}>
+        <FormField
+          control={form.control}
+          name="primarySaleRecipient"
+          render={({ field }) => (
+            <FormItem>
+              <FormDescription>
+                The wallet address that should receive the revenue from initial
+                sales of the assets.
+              </FormDescription>
+              <FormControl>
+                <Input
+                  placeholder="0x..."
+                  {...field}
+                  disabled={!props.isOwnerAccount}
+                />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+
+        <div className="mt-4 flex justify-end">
+          <Button
+            size="sm"
+            className="min-w-24 gap-2"
+            disabled={updateMutation.isPending || !props.isOwnerAccount}
+            type="submit"
+          >
+            {updateMutation.isPending && <Spinner className="size-4" />}
+            Update
+          </Button>
+        </div>
+      </form>{" "}
+    </Form>
+  );
+}
+
+function MintNFTSection(props: {
+  mint: (values: MintFormValues) => Promise<void>;
+  isErc721: boolean;
+  isBatchMetadataInstalled: boolean;
+  isSequentialTokenIdInstalled: boolean;
+}) {
+  const form = useForm<MintFormValues>({
```

Comment:
Add form validation here

Reply:
Can you clarify what you mean by "form validation" here? Since it's using pure typescript typings here, the usual zod resolver can't be used here

Reply:
why can't we add zod resolver here?